src/download.ts

```diff
@@ -612,3 +695,46 @@ function validateResolveHnsResponse(response: AxiosResponse): void {
     );
   }
 }
+
+/**
+ * Validates the registry proof response.
+ *
+ * @param inputSkylink - The input skylink, required to validate the proof.
+ * @param dataLink - The returned data link.
+ * @param proof - The returned proof.
+ * @throws - Will throw if the registry proof header is not present, empty when it shouldn't be, or fails to verify.
+ */
+function validateRegistryProofResponse(inputSkylink: string, dataLink: string, proof?: string): void {
+  let proofArray = [];
+  try {
+    // skyd omits the header if the array is empty.
+    if (proof) {
+      proofArray = JSON.parse(proof);
+      if (!proofArray) {
+        throw new Error("Could not parse 'skynet-proof' header as JSON");
+      }
+    }
+  } catch (err) {
+    throw new Error(`Could not parse 'skynet-proof' header as JSON: ${err}`);
+  }
+
+  if (isSkylinkV1(inputSkylink)) {
+    if (inputSkylink !== dataLink) {
+      throw new Error("Expected returned skylink to be the same as input data link");
+    }
+    // If input skylink is not an entry link, no proof should be present.
+    if (proofArray.length > 0) {
+      throw new Error("Expected 'skynet-proof' header to be empty for data link");
+    }
+    // Nothing else to do for data links, there is no proof to validate.
+    return;
+  }
```

Comment:
we should assert that proof header is not set for skylink v1, checking `proofArray` wouldn't then make sense because in the first try-catch block you already check for if(proof)